pkg/resourcemanager/mysql/mysqluser/mysqluser.go

```diff
@@ -0,0 +1,204 @@
+
+// MSecretUsernameKey is the username key in secret
+const MSecretUsernameKey = "username"
+
+// MSecretPasswordKey is the password key in secret
+const MSecretPasswordKey = "password"
+
+//MySqlUserManager for mysqluser manager
+type MySqlUserManager struct {
+	SecretClient secrets.SecretClient
+	Scheme       *runtime.Scheme
+}
+
+//NewMySqlUserManager creates a new NewMySqlUserManager
+func NewMySqlUserManager(secretClient secrets.SecretClient, scheme *runtime.Scheme) *MySqlUserManager {
+	return &MySqlUserManager{
+		SecretClient: secretClient,
+		Scheme:       scheme,
+	}
+}
+
+// GetDB retrieves a database
+func (s *MySqlUserManager) GetDB(ctx context.Context, resourceGroupName string, serverName string, databaseName string) (db mysql.Database, err error) {
+	dbClient := mysqldatabase.GetMySQLDatabasesClient()
+	return dbClient.Get(
+		ctx,
+		resourceGroupName,
+		serverName,
+		databaseName,
+	)
+}
+
+// ConnectToSqlDb connects to the SQL db using the given credentials
+func (s *MySqlUserManager) ConnectToSqlDb(ctx context.Context, drivername string, fullserver string, database string, port int, user string, password string) (*sql.DB, error) {
+
+	connString := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?tls=skip-verify", user, password, fullserver, port, database)
+
+	db, err := sql.Open(drivername, connString)
+	if err != nil {
+		return db, err
+	}
+
+	err = db.Ping()
+	if err != nil {
+		return db, fmt.Errorf("error ping the mysql db:  %v", err)
+	}
+
+	return db, err
+}
+
+// GrantUserRoles grants roles to a user for a given database
+func (s *MySqlUserManager) GrantUserRoles(ctx context.Context, user string, server string, roles []string, db *sql.DB) error {
+	var errorStrings []string
+	if err := helpers.FindBadChars(user); err != nil {
+		return fmt.Errorf("Problem found with username: %v", err)
+	}
+
+	for _, role := range roles {
+		tsql := fmt.Sprintf("GRANT %s TO '%s'@'%s'", role, user, server)
+
+		if err := helpers.FindBadChars(role); err != nil {
+			return fmt.Errorf("Problem found with role: %v", err)
+		}
+
+		_, err := db.ExecContext(ctx, tsql)
+		if err != nil {
+			errorStrings = append(errorStrings, err.Error())
+		}
+	}
+
+	if len(errorStrings) != 0 {
+		return fmt.Errorf(strings.Join(errorStrings, "\n"))
+	}
+	return nil
+}
+
+// CreateUser creates user with secret credentials
+func (s *MySqlUserManager) CreateUser(ctx context.Context, server string, secret map[string][]byte, db *sql.DB) (string, error) {
+	newUser := string(secret[MSecretUsernameKey])
+	newPassword := string(secret[MSecretPasswordKey])
+
+	// make an effort to prevent sql injection
+	if err := helpers.FindBadChars(newUser); err != nil {
+		return "", fmt.Errorf("Problem found with username: %v", err)
+	}
+	if err := helpers.FindBadChars(newPassword); err != nil {
+		return "", fmt.Errorf("Problem found with password: %v", err)
+	}
+
+	tsql := fmt.Sprintf("CREATE USER IF NOT EXISTS '%s'@'%s' IDENTIFIED BY '%s'", newUser, server, newPassword)
```

Comment:
This approach to building SQL strings is open to SQL Injection...  Can you use the db type in database/sql to build your SQL strings?
https://www.calhoun.io/what-is-sql-injection-and-how-do-i-avoid-it-in-go/

Reply:
@davefellows: Dave, yeah, also tried to avoid SQL injection and use the db.Exec in the commands at first but seems due to the special characters like '' and @, the commands cannot be recognized by mysql. I  understand your concern and we feel the same. Talked with @jananivMS , and we would like to open a new issue https://github.com/Azure/azure-service-operator/issues/1147 to investigate using sql db for the commands. The same issue applies for postgresqluser as well.

Reply:
Thanks. I think it's important we find a solution to this.